checker/src/main/java/org/checkerframework/checker/initialization/InitializationParentAnnotatedTypeFactory.java

```diff
@@ -716,6 +716,7 @@ public Void visitExecutable(AnnotatedExecutableType t, Void p) {
             if (elem.getKind() == ElementKind.CONSTRUCTOR) {
                 AnnotatedDeclaredType returnType = (AnnotatedDeclaredType) t.getReturnType();
                 DeclaredType underlyingType = returnType.getUnderlyingType();
+                // TODO: Make sure we check explicit annotations somewhere.
```

Comment:
I changed this back to `replaceAnnotation`. I am worried that if somebody writes an explicit `@Initialized` on a constructor, nobody will complain. This might replace that with a supertype, which is needed for soundness. For nullness we don't have the same problem, so there using `@NonNull` should be okay. But let's test both of these more.